### PR TITLE
This commit will solve the patch task and handle the errors

### PR DESCRIPTION
**Pull request by AdamG63**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -68,4 +68,80 @@ describe('GET /api/articles/:articles_id', () => {
...
+        };
+        return request(app)
+        .patch(`/api/articles/${ARTICLE_ID}`)
+        .send(articleUpdates)
+        .expect(400)
+        .then(({ body }) => {
+        expect(body.message).toBe('Bad request')
+        })
+    })
+    test('404: responds with not found when passed a articleUpdates that does not exist', () => {
```
Comment (popatre) [inline __tests__/app.test.js]: Is this test not the same as the test on line 108?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -68,4 +68,80 @@ describe('GET /api/articles/:articles_id', () => {
...
+        const ARTICLE_ID = 99
+        const articleUpdates = {
```
Comment (popatre) [inline __tests__/app.test.js]: TEST Missing 

- [ ] 200/400 inc_vote key missing - either respond with an error or make no change.

**Review by popatre — CHANGES_REQUESTED** [review]
Just need one test removing, one test adding

File: __tests__/app.test.js [hunk 3]
```diff
@@ -131,17 +131,18 @@ describe('PATCH /api/articles/:articles_id', () => {
         expect(body.message).toBe('Bad request')
         })
     })
-    test('404: responds with not found when passed a articleUpdates that does not exist', () => {
-        const ARTICLE_ID = 99
+    test('400: responds with not found when passed a missing inc_votes', () => {
+        const ARTICLE_ID = 1
         const articleUpdates = {
-            inc_vote: 100,
+            inc_votes: []
```
Comment (popatre) [inline __tests__/app.test.js]: The key isnt missing, you've just changed the value. This is probably the same test are one of your eariler ones, but with a different value. 
Key missing would look like `  const articleUpdates = {} or   const articleUpdates={inc: 10}`

They dont have the inc_votes key.

**Review by popatre — CHANGES_REQUESTED** [review]

File: app.js [hunk 4]
```diff
@@ -1,18 +1,30 @@
...
 
+app.patch('/api/articles/:article_id', patchArticleById)
+
 app.use((err, req, res, next) => {
    if(err.code === '22P02'){
        res.status(400).send({message: 'Bad request'})
    } else { 
        next(err) }
 })
+
```
Comment (mallionaire) [inline app.js:20]: Try to remember DRY, these two error handlers do a ver similar job, can you think of a refactor that would make it less repetitive?

**Review by mallionaire — CHANGES_REQUESTED** [review]
Have left you a lil comment about your error handlers, please take a look then submit another PR for review

**Review by popatre — APPROVED** [review]
Looks good

**Pull request merged**